src/components/Chart/AreaChart.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import C3Chart from 'react-c3js';
+import ClassNames from 'classnames';
+
+import 'patternfly/dist/js/patternfly-settings';
+import 'patternfly/dist/js/patternfly-settings-charts';
+
+const c3ChartDefaults = window.patternfly.c3ChartDefaults();
+
+/**
+ * AreaChart Component for Patternfly React
+ */
+class AreaChart extends React.Component {
+  constructor(props) {
+    // Check if user set chart type, else set it to default value.
+    if (props.data && !props.data.type) {
+      props.data.type = 'area';
+    }
+    super(props);
```

Comment:
I don't think it's a good idea to manipulate the props at the constructor level.
React components have a life cycle that starts from the props and then when props are getting updated, the component will re-render itself (without calling the constructor again, you have a method called `componentWillReceiveProps(nextProps)`).

Usually, we would want to implement react component as a class when they have inner state management.
I would suggest converting those components to be a stateless functional component.

example:
```js
const AreaChart = ({ className, data, ...props }) => {
  if (data && !data.type) {
    data.type = 'area';
  }
  
  const className = ClassNames(className, {
    'area-chart-pf': true,
  });
  
  return <C3Chart className={className} data={data} {...props} />;
}
```

React life cycle
https://engineering.musefind.com/react-lifecycle-methods-how-and-when-to-use-them-2111a1b692b1
State vs Functional components
https://reactjs.org/docs/components-and-props.html

Reply:
fixed by deriving c3chart, please re-review ...